Python/peephole.c

```diff
@@ -2,536 +2,16 @@
-                    h++;
-                }
-                if (h > i + 1) {
-                    fill_nops(codestr, i + 1, h);
-                    nexti = find_op(codestr, codelen, h);
-                }
-                break;
-        }
-    }
-
-    /* Fixup lnotab */
-    for (i = 0, nops = 0; i < codelen; i++) {
-        size_t block = (size_t)i - nops;
-        /* cannot overflow: codelen <= INT_MAX */
-        assert(block <= UINT_MAX);
-        /* original code offset => new code offset */
-        blocks[i] = (unsigned int)block;
-        if (_Py_OPCODE(codestr[i]) == NOP) {
-            nops++;
-        }
-    }
-    cum_orig_offset = 0;
-    last_offset = 0;
-    for (i=0 ; i < tabsiz ; i+=2) {
-        unsigned int offset_delta, new_offset;
-        cum_orig_offset += lnotab[i];
-        assert(cum_orig_offset % sizeof(_Py_CODEUNIT) == 0);
-        new_offset = blocks[cum_orig_offset / sizeof(_Py_CODEUNIT)] *
-                sizeof(_Py_CODEUNIT);
-        offset_delta = new_offset - last_offset;
-        assert(offset_delta <= 255);
-        lnotab[i] = (unsigned char)offset_delta;
-        last_offset = new_offset;
-    }
-
-    /* Remove NOPs and fixup jump targets */
-    for (op_start = i = h = 0; i < codelen; i++, op_start = i) {
-        j = _Py_OPARG(codestr[i]);
-        while (_Py_OPCODE(codestr[i]) == EXTENDED_ARG) {
-            i++;
-            j = j<<8 | _Py_OPARG(codestr[i]);
-        }
-        opcode = _Py_OPCODE(codestr[i]);
-        switch (opcode) {
-            case NOP:continue;
-
-            case JUMP_ABSOLUTE:
-            case POP_JUMP_IF_FALSE:
-            case POP_JUMP_IF_TRUE:
-            case JUMP_IF_FALSE_OR_POP:
-            case JUMP_IF_TRUE_OR_POP:
-            case JUMP_IF_NOT_EXC_MATCH:
-                j = blocks[j / sizeof(_Py_CODEUNIT)] * sizeof(_Py_CODEUNIT);
-                break;
-
-            case FOR_ITER:
-            case JUMP_FORWARD:
-            case SETUP_FINALLY:
-            case SETUP_WITH:
-            case SETUP_ASYNC_WITH:
-                j = blocks[j / sizeof(_Py_CODEUNIT) + i + 1] - blocks[i] - 1;
-                j *= sizeof(_Py_CODEUNIT);
-                break;
-        }
-        Py_ssize_t ilen = i - op_start + 1;
-        if (instrsize(j) > ilen) {
-            goto exitUnchanged;
-        }
-        /* If instrsize(j) < ilen, we'll emit EXTENDED_ARG 0 */
-        if (ilen > 4) {
-            /* Can only happen when PyCode_Optimize() is called with
-               malformed bytecode. */
-            goto exitUnchanged;
-        }
-        write_op_arg(codestr + h, opcode, j, (int)ilen);
-        h += ilen;
-    }
-    assert(h + (Py_ssize_t)nops == codelen);
-
-    PyMem_Free(blocks);
-    code = PyBytes_FromStringAndSize((char *)codestr, h * sizeof(_Py_CODEUNIT));
-    PyMem_Free(codestr);
-    return code;
-
- exitError:
-    code = NULL;
-
- exitUnchanged:
-    Py_XINCREF(code);
-    PyMem_Free(blocks);
-    PyMem_Free(codestr);
+    Py_INCREF(code);
```

Comment:
You can use `Py_UNUSED` for the ones that are not `code`

Reply:
done